src/extension-support/extension-manager.js

```diff
@@ -133,6 +133,25 @@ class ExtensionManager {
         });
     }
 
+    /**
+    * regenerate blockinfo for any loaded extensions
+    */
+    refreshBlocks () {
+        const extensionsInfo = {};
+        this._loadedExtensions.forEach(extensionURL => {
+            if (builtinExtensions.hasOwnProperty(extensionURL)) {
+                const extension = builtinExtensions[extensionURL];
+                const extensionInstance = new extension(this.runtime);
```

Comment:
There's no record of the service name assigned for the built-in extension, so a new instance was needed.  Perhaps `_loadedExtensions` should be a `new Map()` instead of Set, and then extensions get registered with the serviceName as value: `_loadedExtensions.set(extensionURL, serviceName)`assuming that the serviceName can be available.

Reply:
@cwillisf Any thoughts on this? It seems like we should have a way to get the existing instance?

Reply:
I'm in favor of the `Map` option that @chrisgarrity mentioned. I think the easiest way to accomplish that would be for `_registerInternalExtension` to return a Promise for the service name instead of an empty promise. That way the `then` in `loadExtensionURL` could become `...then(serviceName => {...` and we'd have both the service name and the extension URL/ID in the place where we add a `_loadedExtensions` entry.

Getting this working will be more tricky with non-built-in extensions, but right now those aren't being added to `_loadedExtensions` at all so this change wouldn't make it any worse.